core/src/main/java/org/elasticsearch/index/query/HasParentQueryBuilder.java

```diff
@@ -55,7 +55,7 @@ public HasParentQueryBuilder scoreType(String scoreType) {
     /**
      * Sets inner hit definition in the scope of this query and reusing the defined type and query.
      */
-    public HasParentQueryBuilder innerHit(QueryInnerHitBuilder innerHit) {
+    public HasParentQueryBuilder innerHit(QueryInnerHits innerHit) {
```

Comment:
shall we add this move to QueryInnerHits to `migrate_query_refactoring.asciidoc` given that it breaks the java api?